homeassistant/components/overkiz/climate_entities/hitachi_air_to_air_heat_pump_ovp.py

```diff
@@ -0,0 +1,358 @@
+        if (
+            main_op_state := self.device.states[OverkizState.OVP_MAIN_OPERATION]
+        ) and main_op_state.value_as_str:
+            if main_op_state.value_as_str.lower() == OverkizCommandParam.OFF:
+                return HVACMode.OFF
+
+        if (
+            mode_change_state := self.device.states[OverkizState.OVP_MODE_CHANGE]
+        ) and mode_change_state.value_as_str:
+            # The OVP protocol has 'auto cooling' and 'auto heating' values
+            # that are equivalent to the HLRRWIFI protocol without spaces
+            sanitized_value = mode_change_state.value_as_str.replace(" ", "").lower()
+            return OVERKIZ_TO_HVAC_MODES[sanitized_value]
+
+        return HVACMode.OFF
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set new target hvac mode."""
+        if hvac_mode == HVACMode.OFF:
+            await self._global_control(main_operation=OverkizCommandParam.OFF)
+        else:
+            await self._global_control(
+                main_operation=OverkizCommandParam.ON,
+                hvac_mode=HVAC_MODES_TO_OVERKIZ[hvac_mode],
+            )
+
+    @property
+    def fan_mode(self) -> str | None:
+        """Return the fan setting."""
+        if (
+            state := self.device.states[OverkizState.OVP_FAN_SPEED]
+        ) and state.value_as_str:
+            return OVERKIZ_TO_FAN_MODES[state.value_as_str]
+
+        return None
+
+    async def async_set_fan_mode(self, fan_mode: str) -> None:
+        """Set new target fan mode."""
+        await self._global_control(fan_mode=FAN_MODES_TO_OVERKIZ[fan_mode])
+
+    @property
+    def swing_mode(self) -> str | None:
+        """Return the swing setting."""
+        if (state := self.device.states[OverkizState.OVP_SWING]) and state.value_as_str:
+            return OVERKIZ_TO_SWING_MODES[state.value_as_str]
+
+        return None
+
+    async def async_set_swing_mode(self, swing_mode: str) -> None:
+        """Set new target swing operation."""
+        await self._global_control(swing_mode=SWING_MODES_TO_OVERKIZ[swing_mode])
+
+    @property
+    def target_temperature(self) -> int | None:
+        """Return the target temperature."""
+        if (
+            temperature := self.device.states[OverkizState.CORE_TARGET_TEMPERATURE]
+        ) and temperature.value_as_int:
+            return temperature.value_as_int
+
+        return None
+
+    @property
+    def current_temperature(self) -> int | None:
+        """Return current temperature."""
+        if (
+            state := self.device.states[OverkizState.OVP_ROOM_TEMPERATURE]
+        ) and state.value_as_int:
+            return state.value_as_int
+
+        return None
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new temperature."""
+        temperature = cast(float, kwargs.get(ATTR_TEMPERATURE))
+        await self._global_control(target_temperature=int(temperature))
```

Comment:
```suggestion
        await self._global_control(target_temperature=int(kwargs.get(ATTR_TEMPERATURE)))
```
No need to cast before creating a new int with it